File: scripts/convert_key.py
#!/usr/bin/env python3
import bech32
import binascii
import sys

bech32_key = sys.argv[1] if len(sys.argv) > 1 else 'suiprivkey1qrlgwn3vrmm7w0lpjpl66jmdyxsx2fhk559vwqknt943tk3ntjg4v06hk9t'

try:
    # Decode bech32
    decoded = bech32.decode(bech32_key)
    
    # Convert from 5-bit to 8-bit words, skip version byte
    converted = bech32.convertbits(decoded[1], 5, 8, False)
    
    # Skip first byte (version) and convert to hex
    hex_key = binascii.hexlify(bytes(converted[1:])).decode()
    
    print(hex_key)
except Exception as e:
    print(f"Error: {e}", file=sys.stderr)
    sys.exit(1)

File: scripts/convert-key.js
// Convert Sui bech32 private key to hex format
import { bech32 } from 'bech32';

const bech32Key = process.argv[2] || 'suiprivkey1qrlgwn3vrmm7w0lpjpl66jmdyxsx2fhk559vwqknt943tk3ntjg4v06hk9t';

try {
  const decoded = bech32.decode(bech32Key);
  // Skip the first word (version byte) and convert to hex
  const hexKey = Buffer.from(bech32.toWords(Buffer.from(decoded.words.slice(1)))).toString('hex');
  console.log(hexKey);
} catch (error) {
  console.error('Error converting key:', error.message);
  process.exit(1);
}
